### ISSUE TITLE
`azuredevops_git_repository` - The plugin encountered an error, and failed to respond to the │ plugin.(*GRPCProvider).ReadDataSource call.

### ISSUE THREAD

**Issue by Douglas0n** [microsoft/terraform-provider-azuredevops#1099]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

* Terraform v1.8.2
on linux_amd64

 * terraform-provider-azuredevops_v1.1.1 

### Affected Resource(s)

* `azuredevops_git_repository`

### Terraform Configuration Files

```hcl
data "azuredevops_project" "project" {
  name = var.project
}

data "azuredevops_git_repository" "repo" {
  project_id = data.azuredevops_project.project.id
  name       = var.repo_name
}
```

### Debug Output

https://gist.github.com/Douglas0n/c8c4118aaf218ec7ac9832c5d6bf9657

### Panic Output

https://gist.github.com/Douglas0n/a1d26e5da98e43987708a58233392f45

### Expected Behavior

Inform that the repository does not exist in the team project 

### Actual Behavior

Error: The terraform-provider-azuredevops_v1.1.1 plugin crashed!

### Steps to Reproduce

1. var.project = Any Azure DevOps Team Project
2. var.repo_name = A repository not present in this Team Project
3. `terraform apply`

**Comment by xuzhang3** [comment]
Fixed in #1065

**Comment by Douglas0n** [comment]
I've found that this exception occurs when the provided repository isn't present in the team project.
